### PR TITLE
8324125: Improve class initialization barrier in TemplateTable::_new for RISC-V

### PR DESCRIPTION
**Pull request by zifeihan**
Hi, This RISC-V Port implementation for https://github.com/openjdk/jdk/pull/17006,

### Testing:

- [x]  Run tier1-3 tests on qemu 8.1.0 with UseRVV (fastdebug)
- [x]  Run tier1-3 tests with SiFive unmatched (release)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8324125](https://bugs.openjdk.org/browse/JDK-8324125): Improve class initialization barrier in TemplateTable::_new for RISC-V (**Enhancement** - P4)

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**) ⚠️ Review applies to [ae736225](https://git.openjdk.org/jdk/pull/17548/files/ae73622551cc71d059960105e05467b5bd6902d0)
 * [Robbin Ehn](https://openjdk.org/census#rehn) (@robehn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17548/head:pull/17548` \
`$ git checkout pull/17548`

Update a local copy of the PR: \
`$ git checkout pull/17548` \
`$ git pull https://git.openjdk.org/jdk.git pull/17548/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17548`

View PR using the GUI difftool: \
`$ git pr show -t 17548`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17548.diff">https://git.openjdk.org/jdk/pull/17548.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17548#issuecomment-1909557274)

### CODE REVIEW

File: src/hotspot/cpu/riscv/templateTable_riscv.cpp [hunk 1]
```diff
@@ -3547,11 +3547,9 @@ void TemplateTable::_new() {
   // get InstanceKlass
   __ load_resolved_klass_at_offset(x14, x13, x14, t0);
 
-  // make sure klass is initialized & doesn't have finalizer
-  // make sure klass is fully initialized
-  __ lbu(t0, Address(x14, InstanceKlass::init_state_offset()));
-  __ sub(t1, t0, (u1)InstanceKlass::fully_initialized);
-  __ bnez(t1, slow_case);
+  // make sure klass is initialized
+  assert(VM_Version::supports_fast_class_init_checks(), "Optimization requires support for fast class initialization checks");
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/templateTable_riscv.cpp:3551]: Nit: better to put the msg string on a separate line.

**Review by robehn — APPROVED** [review]
Thanks!

**Pull request closed without merging**